File: src/types/Habit.ts
export type Habit = {
    id: string;
    title: string;
    icon: string;
};

export type HabitDTO = {
    id: string;
    title: string;
    icon: string;
}

export type CreateHabitPayload = {
   profileId: string;
   title: string;
   icon: string;
}

export type CreateHabitInput = {
    profile_id: string;
    title: string;
    icon: string;
}

export type EditHabitPayload = {
    profileId: string;
    habitId: string;
    title: string;
    icon: string;
}

export type EditHabitInput = {
    profile_id: string;
    habit_id: string;
    title: string;
    icon: string;
}

export type DeleteHabitPayload = {
    profileId: string;
    habitId: string;
}

export type DeleteHabitInput = {
    profile_id: string;
    habit_id: string;
}

export type ListHabitsPayload = {
    profileId: string;
}

export type ListHabitsInput = {
    profile_id: string;
}

File: src-tauri/src/models/habit.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Deserialize)]
pub struct CreateHabitInput {
    pub profile_id: String,
    pub title: String,
    pub icon: String,
}

#[derive(Debug, Deserialize)]
pub struct EditHabitInput {
    pub profile_id: String,
    pub habit_id: String,
    pub title: String,
    pub icon: String,
}

#[derive(Debug, Deserialize)]
pub struct DeleteHabitInput {
    pub profile_id: String,
    pub habit_id: String,
}

#[derive(Debug, Deserialize)]
pub struct ListHabitsInput {
    pub profile_id: String,
}

#[derive(Debug, Serialize)]
pub struct Habit {
    pub id: String,
    pub profile_id: String,
    pub title: String,
    pub icon: String,
    pub created_at: String,
    pub updated_at: Option<String>,
    pub deleted_at: Option<String>,
}
